src/common/PreviewUtils.ts

```diff
@@ -0,0 +1,151 @@
+        );
+    }
+
+    public static prefixRouteIfNeeded(compName: string): string {
+        if (compName.toLowerCase().startsWith('c/')) {
+            return compName;
+        }
+        return 'c/' + compName;
+    }
+
+    public static async validateConfigFileWithSchema(
+        configFileJson: any,
+        schema: object
+    ): Promise<ValidationResult> {
+        try {
+            const ajv = new Ajv({ allErrors: true });
+            const result = await ajv.validate(schema, configFileJson);
+            const hasError = ajv.errors ? ajv.errors.length > 0 : false;
+            const errorText = ajv.errors ? ajv.errorsText() : '';
+            const isValid = result === true && hasError === false;
+            return Promise.resolve({
+                errorMessage: errorText,
+                passed: isValid
+            });
+        } catch (err) {
+            return Promise.resolve({
+                errorMessage: err,
+                passed: false
+            });
+        }
+    }
+
+    public static getConfigFileAsJson(file: string): any {
+        const fileContent = fs.readFileSync(file, 'utf8');
+        const json = JSON.parse(fileContent);
+        return json;
+    }
+
+    public static getAppLaunchArguments(
+        json: any,
+        platform: string,
+        targetApp: string
+    ): Map<string, string> {
+        const launchArguments: Map<string, string> = new Map();
+
+        const appDefinition = PreviewUtils.getAppConfig(
+            json,
+            platform,
+            targetApp
+        );
+
+        const args = JsonUtils.getKeyValueIgnoringKeyCase(
+            appDefinition,
+            PreviewUtils.CONFIG_FILE_LAUNCH_ARGUMENTS_KEY
+        ) as [];
+
+        if (args) {
+            args.forEach((arg) => {
+                const name = JsonUtils.getKeyValueIgnoringKeyCase(
+                    arg,
+                    PreviewUtils.CONFIG_FILE_NAME_KEY
+                ) as string;
+
+                const value = JsonUtils.getKeyValueIgnoringKeyCase(
+                    arg,
+                    PreviewUtils.CONFIG_FILE_VALUE_KEY
+                ) as string;
+
+                if (name && value) {
+                    launchArguments.set(name, value);
+                }
+            });
+        }
+
+        return launchArguments;
+    }
+
+    public static getAppLaunchActivity(json: any, targetApp: string): string {
+        const appDefinition = PreviewUtils.getAppConfig(
+            json,
+            CommandLineUtils.ANDROID_FLAG,
+            targetApp
+        );
+
+        const activity = JsonUtils.getKeyValueIgnoringKeyCase(
+            appDefinition,
+            PreviewUtils.CONFIG_FILE_ACTIVITY_KEY
+        ) as string;
+
+        return activity;
+    }
+
+    public static getAppConfig(
+        json: any,
+        platform: string,
+        targetApp: string
+    ): any {
+        const apps = JsonUtils.getKeyValueIgnoringKeyCase(
```

Comment:
We shouldn't ignore case. Case sensitivity is a reasonable constraint in the schema.

Reply:
right now things like `"apps"`  `"ios"`  `"android"`  `"launch_arguments"` are not case sensitive and I wasn't sure if that is intentional or not. So `"iOS"` and `"IOS"` would pass the schema validation.

Perhaps we should update the schema to enforce case sensitivity for everything?